From fffb77d8ede274792d694ee10553dd5e37c97b23 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 15 May 2017 17:25:31 +0800
Subject: [PATCH 1784/5242] MLK-14861-1 mmc: sdhci: make DDR50 tuning
 optionally

commit  587522e9bdc72789e170d66cf10b84377555f342 from
https://source.codeaurora.org/external/imx/linux-imx.git

commit 70f2d20917bc ("MLK-14884 mmc: sdhci: make DDR50 tuning optionally")
want to make DDR50 card tuning optionally, but the code logic is not
right, the tuning of DDR50 card will also be impacted by the flag
'SDHCI_SDR50_NEEDS_TUNING'. e.g. imx6sl/imx6sx/imx6ul/imx7d default set
USE_TUNING_SDR50, which means on these SoC, DDR50 card still do tuning
even haven't the flag 'SDHCI_DDR50_NEEDS_TUNING'.

This patch fix the logic issue, separate DDR50 and SDR50 card.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mmc/host/sdhci.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index d0fd533..b4dbad6 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -2297,9 +2297,12 @@ int sdhci_execute_tuning(struct mmc_host *mmc, u32 opcode)
 		break;
 
 	case MMC_TIMING_UHS_SDR50:
+		if (host->flags & SDHCI_SDR50_NEEDS_TUNING)
+			break;
+		/* FALLTHROUGH */
+
 	case MMC_TIMING_UHS_DDR50:
-		if ((host->flags & SDHCI_SDR50_NEEDS_TUNING) ||
-		    (host->flags & SDHCI_DDR50_NEEDS_TUNING))
+		if (host->flags & SDHCI_DDR50_NEEDS_TUNING)
 			break;
 		/* FALLTHROUGH */
 
-- 
1.7.9.5

